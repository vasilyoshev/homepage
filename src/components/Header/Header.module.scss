.title {
  flex-grow: 1;
  cursor: pointer;
}
