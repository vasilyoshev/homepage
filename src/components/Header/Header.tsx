import {
  AppBar,
  Button,
  IconButton,
  Toolbar,
  Typography,
} from '@material-ui/core';
import MenuIcon from '@material-ui/icons/Menu';
import { NavLink } from 'react-router-dom';
import styles from './Header.module.scss';

export const Header: React.FC = () => {
  return (
    <AppBar position="sticky">
      <Toolbar>
        <IconButton edge="start" color="inherit">
          <MenuIcon />
        </IconButton>
        <Typography variant="h6" className={styles['title']}>
          <NavLink to="/" activeClassName="selected">
            Homepage Widgets
          </NavLink>
        </Typography>
        <Button color="inherit">
          <NavLink to="/about" activeClassName="selected">
            About
          </NavLink>
        </Button>
      </Toolbar>
    </AppBar>
  );
};
