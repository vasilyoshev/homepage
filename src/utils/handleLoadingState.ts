import axios from 'axios';
import { decrementLoadingCount, incrementLoadingCount } from 'slices';
import store from 'store';

export const handleLoadingState = (): void => {
  axios.interceptors.response.use(
    (response) => {
      setTimeout(() => store.dispatch(decrementLoadingCount()), 200);

      return response;
    },
    () => {
      store.dispatch(decrementLoadingCount());
    },
  );

  axios.interceptors.request.use((request) => {
    store.dispatch(incrementLoadingCount());

    return request;
  });
};
