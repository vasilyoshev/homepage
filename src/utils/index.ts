export * from './handleLoadingState';
export * from './testUtils';
