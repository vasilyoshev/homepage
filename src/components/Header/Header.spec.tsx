import { render } from 'utils';
import { Header } from './Header';

describe('Header', () => {
  it('renders successfully', () => {
    const { asFragment } = render(<Header />);

    expect(asFragment()).toMatchSnapshot();
  });
});
