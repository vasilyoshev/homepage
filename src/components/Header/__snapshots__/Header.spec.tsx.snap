// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders successfully 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionSticky MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <h6
        class="MuiTypography-root title MuiTypography-h6"
      >
        <a
          aria-current="page"
          class="selected"
          href="/"
        >
          Homepage Widgets
        </a>
      </h6>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          <a
            href="/signup"
          >
            SIGN UP
          </a>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </header>
</DocumentFragment>
`;
